### PR TITLE
Closes #231

### LINKED ISSUE #231: Create dataset loader for MedDialog
## Adding a Dataset
- **Name:** MedDialog
- **Description:** The MedDialog dataset contains conversations between doctors and patients. It has 1.1 million dialogues in Chinese and 0.26 million dialogues in English
- **Task:** SENT_CLASS
- **Paper:** https://arxiv.org/abs/2004.03329
- **Data:** https://github.com/UCSD-AI4H/Medical-Dialogue-System
- **License:** Public for Research

### PR DESCRIPTION
**Pull request by j-chim**
Please name your PR after the issue it closes. You can use the following line: "Closes #ISSUE-NUMBER" where you replace the ISSUE-NUMBER with the one corresponding to your dataset.

If the following information is NOT present in the issue, please populate:

- **Name:** *name of the dataset*
- **Description:** *short description of the dataset (or link to social media or blog post)*
- **Paper:** *link to the dataset paper if available*
- **Data:** *link to the online home of the dataset*

### Checkbox

- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `biodatasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_BIGBIO_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `BigBioConfig` for the source schema and one for a bigbio schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_bigbio biodatasets/my_dataset/my_dataset.py`.

### CODE REVIEW

File: biodatasets/meddialog/meddialog.py [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+    BIGBIO_VERSION = datasets.Version(_BIGBIO_VERSION)
+
+    BUILDER_CONFIGS = [
+        # Source schemas
+        BigBioConfig(
+            name="meddialog_en_source",
+            version=SOURCE_VERSION,
+            description="MedDialog source schema",
+            schema="source",
+            subset_id="meddialog",
```
Comment (galtay) [inline biodatasets/meddialog/meddialog.py:97]: I think we should make each language a subset id 
```suggestion
            subset_id="meddialog_en",
```

File: biodatasets/meddialog/meddialog.py [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+        ),
+        BigBioConfig(
+            name="meddialog_zh_source",
+            version=SOURCE_VERSION,
+            description="MedDialog source schema",
+            schema="source",
+            subset_id="meddialog",
```
Comment (galtay) [inline biodatasets/meddialog/meddialog.py:104]:
```suggestion
            subset_id="meddialog_zh",
```

File: biodatasets/meddialog/meddialog.py [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+        #    subset_id="meddialog_en",
+        # ),
+        # BigBioConfig(
+        #    name="meddialog_zh_bigbio_text",
+        #    version=BIGBIO_VERSION,
+        #    description="MedDialog simplified BigBio schema",
+        #    schema="bigbio_text",
+        #    subset_id="meddialog_zh",
+        # ),
+    ]
```
Comment (galtay) [inline biodatasets/meddialog/meddialog.py:121]: good idea, lets implement these.

**Review by galtay — CHANGES_REQUESTED** [review]
Thanks for helping us work through the task/schema issues. Lets try this as a text classification task with each (utterance, speaker) tuple as a (text, label) pair.

**Comment by j-chim** [comment]
Thanks @galtay, have incorporated the comments. 

The main concern I have is with the missing descriptions. In this dataset, for English only, there's a "descriptions" that provides context for the utterances. If there is a way to add an optional context field it would be great, e.g., treat each classification instance as a (text, label, context). Have added this in the suggestions channel.

**Comment by galtay** [comment]
Thanks for the suggestions in the discord channel! I think we should definitely keep those schema updates in mind for future iterations / hackathons. I think the source schema leaves a lot of options open for users of this dataset loader. For the moment, I'd like to merge this with the text classification as implemented and revisit when we have a more suitable schema. Does that work for you?

**Comment by j-chim** [comment]
> Thanks for the suggestions in the discord channel! I think we should definitely keep those schema updates in mind for future iterations / hackathons. I think the source schema leaves a lot of options open for users of this dataset loader. For the moment, I'd like to merge this with the text classification as implemented and revisit when we have a more suitable schema. Does that work for you?

Yes this sounds great, thanks @galtay

**Review by galtay — APPROVED** [review]

**Comment by galtay** [comment]
this PR got caught in the middle of a schema change we made for text classification  but I will merge this and then update it.

**Pull request merged**